src/config/configure.in

```diff
@@ -2522,6 +2529,38 @@ then
    dnl CFLAGS=${CXXFLAGS}
 fi
 
+dnl *********************************************************************
+dnl * Auto-enable Umpire for GPU backends (before AC_DEFINEs)
+dnl *********************************************************************
+if test "x$hypre_user_disabled_umpire" != "xyes"
+then
+  if test "x$hypre_using_umpire" != "xyes"
+  then
+    if test "x$hypre_using_cuda" = "xyes" || test "x$hypre_using_hip" = "xyes" || test "x$hypre_using_sycl" = "xyes"
+    then
+      hypre_using_umpire=yes
+      AC_MSG_NOTICE([Enabling Umpire automatically for GPU build (CUDA/HIP/SYCL) for performance and allocator features. Use --without-umpire to opt out.])
+    fi
+  fi
+fi
+
+dnl Warn if Umpire explicitly disabled while using a GPU backend
+if test "x$hypre_user_disabled_umpire" = "xyes"
+then
+  if test "x$hypre_using_cuda" = "xyes" || test "x$hypre_using_hip" = "xyes" || test "x$hypre_using_sycl" = "xyes"
+  then
+    AC_MSG_WARN([===============================================================])
```

Comment:
There are too many brackets around this warning.  We can tone it down to just one set.